android/autodispose-android/src/test/java/com/uber/autodispose/android/internal/MainThreadDisposableTest.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.uber.autodispose.android.internal;
+
+import com.uber.autodispose.android.AutoDisposeAndroidPlugins;
+import io.reactivex.functions.BooleanSupplier;
+import org.junit.Test;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.junit.Assert.assertTrue;
+
+public final class MainThreadDisposableTest {
+
+  @Test public void onDisposeRunsSyncWhenMainThreadSkipped() {
+    AutoDisposeAndroidPlugins.setOnCheckMainThread(new BooleanSupplier() {
+      @Override
+      public boolean getAsBoolean() {
+        return true;
+      }
+    });
+
+    final AtomicBoolean called = new AtomicBoolean();
+
+    new MainThreadDisposable() {
+      @Override
+      protected void onDispose() {
+        called.set(true);
+      }
+    }.dispose();
+
+    assertTrue(called.get());
+    AutoDisposeAndroidPlugins.setOnCheckMainThread(null);
+  }
+
+  @Test
+  public void onDisposeFailsWhenMainThreadCheckNotSet() {
+    try {
+      new MainThreadDisposable() {
+        @Override
+        protected void onDispose() { }
+      }.dispose();
+      throw new AssertionError("Expected to fail before this due to Looper not being stubbed!");
+    } catch (RuntimeException e) {
```

Comment:
Could we make this more specific? Since this is a super specific test case, I'm kind of fine with having something specific. Even just something simple like "contains "stub"" or something simple to smoke test it

Reply:
nvm, saw it's copied